src/lib/warn-upload-overwrite.js

```diff
@@ -94,16 +107,30 @@ const getFormHash = (doc, xml, properties) => {
   return crypt.digest('base64');
 };
 
-const getDocHash = originalDoc => {
+const getDocHash = async (db, originalDoc) => {
   const doc = JSON.parse(JSON.stringify(originalDoc)); // clone doc
   delete doc._rev;
   delete doc._attachments;
   const crypt = crypto.createHash('md5');
   crypt.update(JSON.stringify(doc), 'utf8');
+  const compressibleTypes = await getCompressibleTypes();
+  const matchRegex = (pattern, type) => {
+    const rx = new RegExp(pattern.trim());
+    return rx.test(type);
+  };
   if (originalDoc._attachments) {
-    Object.values(originalDoc._attachments).forEach(attachment => {
-      crypt.update(attachment.digest || couchDigest(attachment.data), 'utf8');
-    });
+    for (const attachmentName of Object.keys(originalDoc._attachments)) {
+      const attachment = originalDoc._attachments[attachmentName];
+      const attachmentCompressible = compressibleTypes ?
+        compressibleTypes.split(',').some(c => matchRegex(c, attachment.content_type)) :
+        true;
```

Comment:
Pretty sad we're going to have to download all attachments for versions of the app that don't implement the new endpoint :(